shell/browser/browser_mac.mm

```diff
@@ -95,6 +98,26 @@
   [[AtomApplication sharedApplication] setShutdownHandler:std::move(handler)];
 }
 
+gfx::Image Browser::CreateThumbnailFromPath(const base::FilePath& path,
+                                            int width,
+                                            int height) {
+  NSString* ns_path = base::mac::FilePathToNSString(path);
+  NSSize size = NSMakeSize(width, height);
+  CFURLRef cfurl = (__bridge CFURLRef)[NSURL fileURLWithPath:ns_path];
+
+  QLThumbnailRef ql_thumbnail = QLThumbnailCreate(
+      kCFAllocatorDefault, cfurl, CGSizeMake(size.width, size.height), NULL);
+  CGImageRef cg_thumbnail = QLThumbnailCopyImage(ql_thumbnail);
```

Comment:
We should be using `QLThumbnailDispatchAsync` first here to ensure that we don't block the main thread while we wait for quicklook to generate the thumbnail.  This API will need to return a promise.

Refs: https://developer.apple.com/documentation/quicklook/1402621-qlthumbnailcopyimage?language=objc

> If you call this function directly without using QLThumbnailDispatchAsync, this function blocks the calling thread until it finishes computing the thumbnail.